pytest_django/fixtures.py

```diff
@@ -125,19 +125,60 @@ def django_db_setup(
             **setup_databases_args
         )
 
+
     def teardown_database() -> None:
         with django_db_blocker.unblock():
             try:
-                teardown_databases(db_cfg, verbosity=request.config.option.verbose)
+                if django_db_keepdb:
+                    # TODO: this would better belong in core django but that would require us to patch 2 libraries
+
+                    # When using Transactions django truncates table at the end of the test, and then next
+                    # test restores the snapshot (if using serialized_rollback). The issue however arises 
+                    # when we are also using --keep-db/reuse-db flag, since database is left empty at the end
+                    # and next test run does not know how to restore the db.
+                    # To account for that we restore db from snapshot once again at the end of the tests.
+
+                    verbosity = request.config.option.verbose
+                    parallel = 0
+                    for connection, old_name, destroy in db_cfg:
+                        if destroy:
+                            if parallel > 1:
+                                for index in range(parallel):
+                                    connection.creation.destroy_test_db(
+                                        suffix=str(index + 1),
+                                        verbosity=verbosity,
+                                        keepdb=True,
+                                    )
+                            connection.creation.destroy_test_db(old_name, verbosity, True)
```

Comment:
i think this is only needed if _test_serialized_contents is set. Otherwise there were no transactional tests involved, so we can save some time and just do nothing